### PR TITLE
connect to api

### PR DESCRIPTION
**Pull request by wolterbwambale**
###  In this project, I have implemented the following:-

- [ ] Make use of the [Bookstore API]
- [ ] Fetch your books from the Bookstore API after the component has rendered
- [ ] Refactor addbook and remove book features

### CODE REVIEW

File: src/components/BookList.js [hunk 1]
```diff
@@ -1,18 +1,18 @@
 import React from 'react';
 import { useSelector } from 'react-redux';
-import Books from './Book';
+import Book from './Book';
```
Comment (topeogunleye) [inline src/components/BookList.js:5]: Hi, you have a good start with your code, but there are some changes that you need to make to meet the project requirement. You are missing the `useEffect` hook and the `fetchBooks` function in your `BookList` component. The `useEffect` hook is used to perform side effects, such as fetching data from an API, in a React component. The `fetchBooks` function is presumably the action creator that dispatches an action to fetch the books from an API and update the state. You need to import these dependencies and use them in your component. For example:

```js
import React, { useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import Book from './Book';
import { fetchBooks } from '../actions/bookActions';

const BookList = () => {
  const booklist = useSelector((state) => state.book.booklist);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchBooks());
  }, [dispatch]);

  if (!booklist || booklist.length === 0) {
    return <p>No books available.</p>;
  }

  return (
    <>
      {booklist.map((book) => (
        <Book key={book.item_id} item={book} />
      ))}
    </>
  );
};

export default BookList;

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @wolterbwambale, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Try-Again

> ## (Highlights) Good Points: 💪 

- You have imported createAsyncThunk from "@reduxjs/toolkit" and used it to create a fetchBooks action creator that makes a GET request to the API using axios and returns the response data. 🙌
- You have added an extraReducers property to your books slice and handled the pending, fulfilled and rejected cases for the fetchBooks action. This ensures that your state reflects the status and result of the async action. 🙌
- You have refactored your addBook and removeBook action creators to use createAsyncThunk as well. They make POST and DELETE requests to the API using axios and return a promise. They also automatically dispatch actions that add or remove a book from the state. 🙌
- You have tested your app in the browser and verified that everything still works as expected. Your app can display, add and remove books from the API without any errors or bugs. 🙌
> * PR has a good title and summary :heavy_check_mark:

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Nothing to mention.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/BookList.js (hunk 1 above)
Comment (wolterbwambale) [reply src/components/BookList.js:5]: @topeogunleye  thank you

**Review by bienvenuushindi — APPROVED** [review]
# STATUS: APPROVED 🟢 

## Hi @wolterbwambale   👋, 

Well done implementing the requested changes 👏
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Highlights 🎯 
:+1: No linter errors✔️
:+1: Good UI design✔️
:+1: Good PR title

### Optional Suggestions



**_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by wolterbwambale** [comment]
@bienvenuushindi  thank youo

**Pull request merged**